spring-components/spring-boot-sparql-web/src/main/java/org/eclipse/rdf4j/http/server/readonly/sparql/SparqlQueryEvaluator.java

```diff
@@ -24,13 +24,14 @@ public interface SparqlQueryEvaluator {
 	 * @param repository      the repository against which the query is to be executed
 	 * @param query           The query to be evaluated
 	 * @param acceptHeader    needed to find the best response format.
-	 * @param defaultGraphUri see {@link Dataset#getDefaultGraphs()}
-	 * @param namedGraphUris  see {@link Dataset#getNamedGraphs()}
+	 * @param defaultGraphUri The graphs that are embedded in the query with FROM {@link Dataset#getDefaultGraphs()}
+	 * @param namedGraphUris  The graphs that are embedded in the query with FROM NAMED see
+	 *                        {@link Dataset#getNamedGraphs()}
 	 * @throws MalformedQueryException If the supplied query is malformed.
 	 * @throws IOException             if there is a problem with the {@link EvaluateResult#getOutputstream()}
 	 * @throws IllegalStateException   if no acceptHeader is present
 	 */
 	void evaluate(EvaluateResult result, Repository repository, String query, String acceptHeader,
-			String defaultGraphUri,
+			String[] defaultGraphUri,
 			String[] namedGraphUris) throws MalformedQueryException, IllegalStateException, IOException;
```

Comment:
This is a change to a public interface. It's not backwards compatible. 

@abrokenjester Do you know why japicmp doesn't complain about this? I get a feeling that japicmp is disabled by default and needs to be enabled in each module.

Reply:
i will change that, so that there are 2 methods. the "old" one can be marked as deprecated.

Reply:
Ok. The new interface method needs to have a default implementation.

Reply:
I do not understand the reason for this. What should the default implementation do ?

Reply:
```
	default void evaluate(EvaluateResult result, Repository repository, String query, String acceptHeader,
			String[] defaultGraphUris,
			String[] namedGraphUris) throws MalformedQueryException, IllegalStateException, IOException {
		String defaultGraphUri = defaultGraphUris != null ? defaultGraphUris[0] : null;
		evaluate(result, repository, query, acceptHeader, defaultGraphUri, namedGraphUris);
	}
```

Reply:
done